options_filter.go

```diff
@@ -23,8 +23,8 @@ func (f *filtersConnectionGater) InterceptPeerDial(p peer.ID) (allow bool) {
 	return true
 }
 
-func (f *filtersConnectionGater) InterceptAccept(_ network.ConnMultiaddrs) (allow bool) {
-	return true
+func (f *filtersConnectionGater) InterceptAccept(connAddr network.ConnMultiaddrs) (allow bool) {
+	return !(*ma.Filters)(f).AddrBlocked(connAddr.RemoteMultiaddr())
```

Comment:
@aarshkshah1992 is this the only place to do this? Don't we need something for at least InterceptSecured if not InterceptUpgraded also?

Reply:
@aschmahmann 

An `InterceptUpgraded` wont do match for us here as all those rejections will happen in `InterceptAccepted` also since we didn't have the notion of rejecting connections at different "lifecycle states" when we had this old filter.

However, there is no harm in doing so and I am okay either ways.

Reply:
Actually @aschmahmann , we might need this for QUIC. Thanks for catching this.

@jalextowle Please can you add a similar behaviour to `InterceptSecured` as well ?

Reply:
@aarshkshah1992 Yep, I added that behavior.